src/addons/addonmessage.cpp

```diff
@@ -224,3 +226,21 @@ qint64 AddonMessage::planDateRetranslationInternal(
 
   return 86400 + secsTo;
 }
+
+void AddonMessage::setBadge(const QString& badge) {
+    m_badge = None;
+    if (badge.isEmpty() || badge == "none") {  // Nothing to do.
+    } else if (badge == "warning") {
+      m_badge = Warning;
+    } else if (badge == "critical") {
+      m_badge = Critical;
+    } else if (badge == "new_update") {
+        m_badge = NewUpdate;
+    } else if (badge == "whats_new") {
+        m_badge = WhatsNew;
+    } else if (badge == "survey") {
+        m_badge = Survey;
+    } else {
+      logger.error() << "Unsupported badge type" << badge;
```

Comment:
mmm do we want to return an error here and reject the addon?

Reply:
I feel like the badge isn't mission critical to the addon and the message would still be worth showing, but I could go either way